services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/common/AttachmentValidator.kt

```diff
@@ -54,50 +52,41 @@ object AttachmentValidator {
             "LPT9",
         )
 
-    fun validate(attachment: MultipartFile) =
-        with(attachment) {
-            logger.info { "Validating file $originalFilename of type: $contentType" }
-            if (contentType.isNullOrBlank() || !validFileName(originalFilename)) {
-                throw AttachmentInvalidException("File '$originalFilename' not supported")
-            }
-        }
-
-    private fun validFileName(fileName: String?): Boolean {
-        if (fileName.isNullOrBlank()) {
-            logger.warn { "Attachment file name null or blank" }
-            return false
-        }
-
-        if (fileName.length > 128) {
-            logger.warn { "File name is too long" }
-            return false
-        }
+    fun validFilename(originalFilename: String?): String {
+        logger.info { "Validating file name $originalFilename" }
 
-        if (!supportedType(fileName)) {
-            logger.warn { "File '$fileName' not supported" }
-            return false
-        }
+        val sanitizedFilename = sanitizeFilename(originalFilename)
+        logger.info { "Sanitized file name to $sanitizedFilename" }
 
-        if (fileName.contains("..")) {
-            logger.warn { "File name contains path traversal characters" }
-            return false
+        if (!validateFileName(sanitizedFilename)) {
+            throw AttachmentInvalidException("File '$sanitizedFilename' not supported")
         }
+        return sanitizedFilename!!
+    }
 
-        if (INVALID_CHARS_PATTERN.matcher(fileName).find()) {
-            logger.warn { "File name contains invalid characters" }
-            return false
-        }
+    private fun sanitizeFilename(filename: String?): String? =
+        filename?.replace(INVALID_CHARS_PATTERN, "_")
 
-        if (RESERVED_NAMES.contains(removeExtension(fileName).uppercase(Locale.getDefault()))) {
-            logger.warn { "File name is reserved" }
-            return false
+    private fun validateFileName(filename: String?): Boolean =
```

Comment:
This could possible renamed to isValidFilename, or just isValidName. It should improve readability when called: !isValid etc. instead of !validate.

Reply:
Yes, that's better. Renamed to `isValidFilename`. Thank you for the suggestion.